velox/exec/tests/ExpandTest.cpp

```diff
@@ -0,0 +1,200 @@
+          .planNode();
+
+  assertQuery(
+      plan,
+      "SELECT k1, k2, count(1), sum(a), max(b) FROM tmp GROUP BY GROUPING SETS ((k1), (k1), (k2))");
+}
+
+TEST_F(ExpandTest, cube) {
+  vector_size_t size = 1'000;
+  auto data = makeRowVector(
+      {"k1", "k2", "a", "b"},
+      {
+          makeFlatVector<int64_t>(size, [](auto row) { return row % 11; }),
+          makeFlatVector<int64_t>(size, [](auto row) { return row % 17; }),
+          makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+          makeFlatVector<std::string>(
+              size, [](auto row) { return std::string(row % 12, 'x'); }),
+      });
+
+  createDuckDbTable({data});
+
+  // Cube.
+  auto plan =
+      PlanBuilder()
+          .values({data})
+          .expand({
+              {"k1", "k2", "a", "b", "0 as group_id_0"},
+              {"k1", "null", "a", "b", "1"},
+              {"null", "k2", "a", "b", "2"},
+              {"null", "null", "a", "b", "3"},
+          })
+          .singleAggregation(
+              {"k1", "k2", "group_id_0"},
+              {"count(1) as count_1", "sum(a) as sum_a", "max(b) as max_b"})
+          .project({"k1", "k2", "count_1", "sum_a", "max_b"})
+          .planNode();
+
+  assertQuery(
+      plan,
+      "SELECT k1, k2, count(1), sum(a), max(b) FROM tmp GROUP BY CUBE (k1, k2)");
+}
+
+TEST_F(ExpandTest, rollup) {
+  vector_size_t size = 1'000;
+  auto data = makeRowVector(
+      {"k1", "k2", "a", "b"},
+      {
+          makeFlatVector<int64_t>(size, [](auto row) { return row % 11; }),
+          makeFlatVector<int64_t>(size, [](auto row) { return row % 17; }),
+          makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+          makeFlatVector<std::string>(
+              size, [](auto row) { return std::string(row % 12, 'x'); }),
+      });
+
+  createDuckDbTable({data});
+
+  // Rollup.
+  auto plan =
+      PlanBuilder()
+          .values({data})
+          .expand(
+              {{"k1", "k2", "a", "b", "0 as group_id_0"},
+               {"k1", "null", "a", "b", "1"},
+               {"null", "null", "a", "b", "2"}})
+          .singleAggregation(
+              {"k1", "k2", "group_id_0"},
+              {"count(1) as count_1", "sum(a) as sum_a", "max(b) as max_b"})
+          .project({"k1", "k2", "count_1", "sum_a", "max_b"})
+          .planNode();
+
+  assertQuery(
+      plan,
+      "SELECT k1, k2, count(1), sum(a), max(b) FROM tmp GROUP BY ROLLUP (k1, k2)");
+}
+
+TEST_F(ExpandTest, countDistinct) {
+  vector_size_t size = 1'000;
+  auto data = makeRowVector(
+      {"a", "b"},
+      {
+          makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+          makeFlatVector<std::string>(
+              size, [](auto row) { return std::string(row % 12, 'x'); }),
+      });
+  createDuckDbTable({data});
+
+  // count distinct.
+  auto plan =
+      PlanBuilder()
+          .values({data})
+          .expand({{"a", "null as b", "1 as group_id_0"}, {"null", "b", "2"}})
```

Comment:
This plan doesn't seem to use group_id_0 column. This is unexpected. Would you double check?

Reply:
@mbasmanova Have updated the test sql using `group_id_0 `column.